trio_run_in_process/process.py

```diff
@@ -182,7 +193,8 @@ def poll(self) -> Optional[int]:
 
     def kill(self) -> None:
         self.send_signal(signal.SIGKILL)
-        self.status = State.FINISHED
+        self.state = State.FINISHED
+        self.returncode = -signal.SIGKILL
```

Comment:
Any thoughts on adding protection to avoid overwriting if the return code is already present  (maybe in a separate PR?)

Reply:
I was going to do that in this PR but then realized this method (as well as `terminate()`) will kill the actual worker process, rendering it unusable without actually flagging it as such. For example, the following will block forever on the second `worker.open(...)` call because the OS process is dead. 

```
        async with open_worker_process() as worker:
            async with worker.open(return7) as proc:
                proc.terminate()
                await trio.sleep(0.1)

            async with worker.open(return7) as proc:
                assert 7 == await proc.wait_result_or_raise()
```

I think we can't simply remove the kill/terminate methods from the object yielded by `worker.open()`, and I don't see how we could have them terminate whatever job the worker is performing without actually terminating the worker, so maybe what we need is to have those methods flag the worker as dead as well

Reply:
Another alternative would be to make `WorkerProcess.open()` private, only to be used in `open_in_process()`. That way `Process` instances are only exposed to callers of `open_in_process()`, in which case we don't have to worry about the `WorkerProcess` in case they call `kill/terminate` as the worker is not exposed and wouldn't be reused. This feels like the best option to me, in fact